service/controller/v11/resource/migration/resource.go

```diff
@@ -0,0 +1,83 @@
+// Package migration provides an operatorkit resource that migrates awsconfig CRs
+// to reference the default credential secret if they do not already.
+// It can be safely removed once all awsconfig CRs reference a credential secret.
+package migration
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/aws-operator/service/controller/v11/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+)
+
+const (
+	Name = "migrationv11"
+
+	AWSConfigNamespace               = "default"
+	CredentialSecretDefaultNamespace = "giantswarm"
+	CredentialSecretDefaultName      = "credential-default"
+)
+
+type Config struct {
+	G8sClient versioned.Interface
+	Logger    micrologger.Logger
+}
+
+type Resource struct {
+	g8sClient versioned.Interface
+	logger    micrologger.Logger
+}
+
+func New(config Config) (*Resource, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	r := &Resource{
+		g8sClient: config.G8sClient,
+		logger:    config.Logger,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return Name
+}
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// If credential secret is set we have nothing to migrate.
+	if customObject.Spec.AWS.CredentialSecret.Name != "" {
+		return nil
+	}
```

Comment:
Will this be used outside of the migration resource? If so then a key function would be cool.

Reply:
I have mixed feeling about that. We don't have key function for setting things and this value is set a few lines later.

This is a migration resource so I don't see anything wrong with it knowing the structure.

Reply:
Yes, good point about setting the value. If this is just used within the migration resource I think its fine without it.

If we need to read the value in other resources a key function can be added later.